src/nethsec/mwan/__init__.py

```diff
@@ -0,0 +1,424 @@
+        raise ValidationError('name', 'unique', name)
+    # generate policy config with corresponding name
+    e_uci.set('mwan3', policy_config_name, 'policy')
+    e_uci.set('mwan3', policy_config_name, 'label', name)
+    changed_config.append(f'mwan3.{policy_config_name}')
+
+    member_names = __add_interfaces(e_uci, interfaces, changed_config)
+
+    e_uci.set('mwan3', policy_config_name, 'use_member', member_names)
+
+    if len(utils.get_all_by_type(e_uci, 'mwan3', 'rule')) == 0:
+        changed_config.append(store_rule(e_uci, 'Default Rule', policy_config_name))
+
+    e_uci.save('mwan3')
+    e_uci.save('network')
+    return changed_config
+
+
+def __fetch_interface_status(interface_name: str) -> str:
+    try:
+        output = (subprocess.run([
+            'ubus',
+            'call',
+            'mwan3',
+            'status',
+            '{"section": "interfaces"}'
+        ], capture_output=True)
+                  .stdout.decode('utf-8'))
+        decoded_output = json.JSONDecoder().decode(output)
+        return decoded_output['interfaces'][interface_name]['status']
+    except:
+        return 'unknown'
+
+
+def __parse_member(e_uci: EUci, member_name: str) -> dict:
+    """
+    Parses a member configuration and returns formatted data.
+    Args:
+        e_uci: EUci instance
+        member_name: member name
+
+    Returns:
+        dict with member data
+    """
+    member_data = e_uci.get_all('mwan3', member_name)
+    return {
+        'name': member_name,
+        'interface': member_data['interface'],
+        'metric': member_data['metric'],
+        'weight': member_data['weight'],
+        'status': __fetch_interface_status(member_data['interface'])
+    }
+
+
+def index_policies(e_uci: EUci) -> list[dict]:
+    """
+    Returns a list of policies with their members, interfaces and metrics/weights.
+    Args:
+        e_uci: EUci instance
+
+    Returns:
+        list of dicts with policy data
+    """
+    data = []
+    policies = utils.get_all_by_type(e_uci, 'mwan3', 'policy')
+    # iterate over policies
+    for policy_name in policies.keys():
+        policy = policies[policy_name]
+        policy_data = {
+            'name': policy_name,
+        }
+        # add label only if present
+        if 'label' in policy:
+            policy_data['label'] = policy['label']
+
+        # add members
+        members = []
+        if 'use_member' in policy:
+            policy_data['members'] = {}
+            for member in policy['use_member']:
+                members.append(__parse_member(e_uci, member))
+
+        # infer policy type by metrics
+        metrics = [int(member['metric']) for member in members]
+        if all(metric == metrics[0] for metric in metrics):
+            policy_data['type'] = 'balance'
+        elif all(metrics.index(metric) == key for key, metric in enumerate(metrics)):
+            policy_data['type'] = 'backup'
+        else:
+            policy_data['type'] = 'custom'
```

Comment:
While this looks good, why not adding a `ns_tag` (or similar) to the policy to track it easily?
Just an idea, feel free to ignore it.

Reply:
I'm currently using the name of the policy/routes/metrics `ns_<name_normalized>` to keep track of which policy is created by APIs or not.
Probably I'll add something like it down the line, however it's a simple prefix check on `name` field